### ISSUE TITLE
LARS in EfficientNet training phase?

### ISSUE THREAD

**Issue by CyFeng16** [tensorflow/tpu#571]
@mingxingtan Accuracy drop occurs when I train EfficientNte from scratch using TPU Pod (v2-512). Are there any LARS-like methods used in training? Or any recommended cloud TPU configuration can be provided? Many thanks.

**Comment by mingxingtan** [comment]
What total batch size did you use and how much accuracy drop did you see?

**Comment by CyFeng16** [comment]
Configuration:
- TPU ACCELERATOR_TYPE: v2-512
- TPU VERSION: 1.15
- batch_size: 32768 (according to [this config](https://github.com/tensorflow/tpu/blob/master/models/official/resnet/configs/cloud/v2-512.yaml))
- train_steps: 13684
lr and other settings stand.

Final results are as following:
- 'loss': 3.7762508
- 'top_1_accuracy': 0.4031372
- 'top_5_accuracy': 0.6648763
- 'global_step': 13684

**Comment by CyFeng16** [comment]
Results, when batch_szie has been set as 16384(2048*8), will be updated soon.

**Comment by CyFeng16** [comment]
@mingxingtan LARS really helps maintain performance when we using large batch (e.g. 32768). Codes refer to [this](https://github.com/tensorflow/tpu/blob/master/models/official/resnet/lars_util.py).

**Comment by CyFeng16** [comment]
https://github.com/tensorflow/tpu/issues/546#issuecomment-542390006

**Comment by mingxingtan** [comment]
Hi @CyFeng16 , could you help try LARS for EfficientNet with large batch size (like 32k)?

**Comment by CyFeng16** [comment]
I have tried it, and fortunately, it worked! However, it turns out that after more than 4k batchsize, the training process is unstable due to tpuBatchNorm.

**Comment by CyFeng16** [comment]
Both 16k and 32k could reach 0.77+ top-1 acc without using AutoAugment methods, although facing NaN frequently.

**Comment by mingxingtan** [comment]
Excellent! Do you mind sending out a PR for LARS+EfficientNet? I  can help debug the NaN issue.

**Comment by CyFeng16** [comment]
@mingxingtan 
I am very willing to contribute here, but I also faced a DDL recently :(
SO, I am just not quite sure I have enough time to clean-up the code from my repository.

**Comment by haossr** [comment]
Quick follow up on this thread. Try to reproduce b0 with batch size from 4096 - 65536 on v2-512
- Current lr scaling (linear, based on 256 batch size) stop working at 16384. 
- Still can reproduce if switching to root square lr scaling:
`scaled_lr = self.p.base_learning_rate * np.sqrt(train_batch_size / 256)`
- With global batch size 65536, the model got converged in less than 4000 steps

**Comment by mingxingtan** [comment]
Very interesting finding. Thank you for the information, @haossr !
